src/ui-components/dashboard/dashboard.component.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FunctionComponent } from "react";
+import styles from "./dashboard.css";
+import Widget from "../widget/widget.component";
+
+export default function Dashboard(props: DashboardProps) {
+  const [widgets, setWidgets] = React.useState([]);
+
+  React.useEffect(() => {
+    //loadDashboardFromConfig(props.dashboardConfig);
+
+    const widgets = [];
+    let widget;
+    props.dashboardConfig.widgets.forEach((widgetConfig, index) => {
+      widget = <Widget widgetConfig={widgetConfig} />;
```

Comment:
I think variable widget is not being used. Could this be a repetition? Also using props.dashboardConfig.widgets.map would have resulted in fewer lines of code to maintain, hence fewer places to make mistakes.

Reply:
replaced with this: 
` setWidgets(
      props.dashboardConfig.widgets.map(widgetConfig => {
        widgetConfig.component = () => <Widget widgetConfig={widgetConfig} />;
        return widgetConfig;
      })
    );`